pkg/antctl/raw/check/cluster/command.go

```diff
@@ -0,0 +1,236 @@
+			{Name: "lib-modules", MountPath: "/lib/modules"},
+			{Name: "os-info", MountPath: "/etc/os-release"},
+		},
+		Tolerations: []corev1.Toleration{
+			{
+				Key:      "node-role.kubernetes.io/control-plane",
+				Operator: "Exists",
+				Effect:   "NoSchedule",
+			},
+			{
+				Key:      "node.kubernetes.io/not-ready",
+				Operator: "Exists",
+				Effect:   "NoSchedule",
+			},
+		},
+		Volumes: []corev1.Volume{
+			{
+				Name: "cni-conf",
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: "/etc/cni/net.d",
+					},
+				},
+			},
+			{
+				Name: "lib-modules",
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: "/lib/modules",
+						Type: hostPathTypePtr("Directory"),
+					},
+				},
+			},
+			{
+				Name: "os-info",
+				VolumeSource: corev1.VolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: "/etc/os-release",
+						Type: hostPathTypePtr("File"),
+					},
+				},
+			},
+		},
+	})
+
+	t.Log("Creating Deployment")
+	_, err := t.client.AppsV1().Deployments(antreaNamespace).Create(ctx, deployment, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to create Deployment: %w", err)
+	}
+
+	t.Log("Waiting for Deployment to become ready")
+	t.waitForDeploymentsReady(ctx, time.Second, podReadyTimeout, deploymentName)
+	if err != nil {
+		return fmt.Errorf("error while waiting for Deployment to become ready: %w", err)
+	}
+	return nil
+}
+
+func hostPathTypePtr(s string) *corev1.HostPathType {
+	v := corev1.HostPathType(s)
+	return &v
+}
+
+func NewTestContext(client kubernetes.Interface, config *rest.Config, clusterName string, o *options) *testContext {
+	return &testContext{
+		client:          client,
+		config:          config,
+		clusterName:     clusterName,
+		antreaNamespace: o.antreaNamespace,
+	}
+}
+
+func (t *testContext) teardown(ctx context.Context, deploymentName, namespace string) error {
+	err := t.client.AppsV1().Deployments(namespace).Delete(ctx, deploymentName, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+	t.Log("Waiting for the deletion of Deployment %s in Namespace %s...", deploymentName, namespace)
+	err = wait.PollUntilContextTimeout(ctx, 2*time.Second, 1*time.Minute, true, func(ctx context.Context) (bool, error) {
+		_, err := t.client.AppsV1().Deployments(namespace).Get(ctx, deploymentName, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			return true, nil
+		}
+		if err != nil {
+			return false, err
+		}
+		return false, nil
+	})
+	if err != nil {
+		return fmt.Errorf("error waiting for Deployment %s to be deleted in Namespace %s: %w", deploymentName, namespace, err)
+	}
+	t.Log("Deployment %s successfully deleted from Namespace %s", deploymentName, namespace)
+	return nil
+}
+
+func (t *testContext) waitForDeploymentsReady(ctx context.Context, interval, timeout time.Duration, deployments ...string) error {
```

Comment:
Yes, Making it a universal function is causing problems in the sequential execution of this function therefore it is producing errors Therefore I kept it as it is

Reply:
can you elaborate a little bit? what do you mean by "problems in the sequential execution of this function"?

Reply:
I've noticed that Sometimes It is getting executed before a deployment is initiated.
Also it sometimes waits for a different deployment instead of the deployment it should wait for.

Reply:
I am confused: how does duplicating the function declaration (https://github.com/antrea-io/antrea/blob/f9345daedbce39e77debe9bf3116397da647e57c/pkg/antctl/raw/check/installation/command.go#L378) help with these issues? As far as I can tell, you would just need to make the namespace a function parameter.

Reply:
Yes thank you, It worked after some code improvements :)